IdentityCore/src/cache/key/MSIDDefaultCredentialCacheKey.m

```diff
@@ -34,38 +35,44 @@ @implementation MSIDDefaultCredentialCacheKey
 
 #pragma mark - Helpers
 
-// kSecAttrService - (<credential_type>-<client_id>-<realm>-<target>)
+// kSecAttrService - (<credential_type>-<client_id>-<realm>-<enrollment_id>-<target>)
 - (NSString *)serviceWithType:(MSIDCredentialType)type
                      clientID:(NSString *)clientId
                         realm:(NSString *)realm
+                 enrollmentId:(NSString *)enrollmentId
                        target:(NSString *)target
 {
     realm = realm.msidTrimmedString.lowercaseString;
     clientId = clientId.msidTrimmedString.lowercaseString;
     target = target.msidTrimmedString.lowercaseString;
+    enrollmentId = enrollmentId.msidTrimmedString.lowercaseString;
 
-    NSString *credentialId = [self credentialIdWithType:type clientId:clientId realm:realm];
+    NSString *credentialId = [self credentialIdWithType:type clientId:clientId realm:realm enrollmentId:enrollmentId];
     NSString *service = [NSString stringWithFormat:@"%@%@%@",
                          credentialId,
                          keyDelimiter,
                          (target ? target : @"")];
     return service;
 }
 
-// credential_id - (<credential_type>-<client_id>-<realm>)
+// credential_id - (<credential_type>-<client_id>-<realm>-<enrollment_id>)
```

Comment:
why are we putting enrollmentId to the middle of service instead of end?

Reply:
Happens to be there, as _service_ consists of generic+realm, and we put _enrollmentId_ in the end of _generic_